domain/robstoll-lib/atrium-domain-robstoll-lib-jvm/src/test/kotlin/ch/tutteli/atrium/assertions/filesystem/SymbolicLinkResolvingSpec.kt

```diff
@@ -0,0 +1,182 @@
+                val innerLink = Files.createSymbolicLink(parent.resolve("__linkTo_${target.fileName}"), target)
+                val innerLinkInGrandparentLink = grandparentlink.resolve(parent.fileName).resolve(innerLink.fileName)
+                val linkToInnerLink = tempFolder.newSymbolicLink(
+                    "__transitive_linkTo_${target.fileName}", innerLinkInGrandparentLink
+                )
+
+                explainForResolvedLink(linkToInnerLink, expectListener)
+                verify(expectListener)(target)
+            }
+        }
+
+        describe("explains correctly") {
+            it("returns the original assertion if no link is involved") {
+                val file = tempFolder.newFile("testFile")
+
+                val resultAssertion = explainForResolvedLink(file, expectListener)
+                expect(resultAssertion).isSameAs(testAssertion)
+            }
+
+            it("adds an explanation for one symbolic link") {
+                val target = tempFolder.tmpDir.resolve("notExisting")
+                val link = tempFolder.newSymbolicLink("link", target)
+
+                val resultAssertion = explainForResolvedLink(link, expectListener)
+                expect(resultAssertion).isA<AssertionGroup>().feature({ p(it::assertions) }) {
+                    get(0).describesLink(link, target)
+                    get(1).isSameAs(testAssertion)
+                }
+            }
+
+            it("adds explanations for multiple symbolic links") {
+                val target = tempFolder.tmpDir.resolve("notExisting")
+                val grandparent = tempFolder.newFolder("__linksgrandparent")
+                val parent = Files.createDirectory(grandparent.resolve("step"))
+                val grandparentlink = tempFolder.newSymbolicLink("__linkTo_grandparent", grandparent)
+                val innerLink = Files.createSymbolicLink(parent.resolve("__linkTo_${target.fileName}"), target)
+                val innerLinkInGrandparentLink = grandparentlink.resolve(parent.fileName).resolve(innerLink.fileName)
+                val linkToInnerLink = tempFolder.newSymbolicLink(
+                    "__transitive_linkTo_${target.fileName}", innerLinkInGrandparentLink
+                )
+
+                val resultAssertion = explainForResolvedLink(linkToInnerLink, expectListener)
+                expect(resultAssertion).isA<AssertionGroup>().feature({ p(it::assertions) }) {
+                    get(0).describesLink(linkToInnerLink, innerLinkInGrandparentLink)
+                    get(1).describesLink(grandparentlink, grandparent)
+                    get(2).describesLink(innerLink, target)
+                    get(3).isSameAs(testAssertion)
+                }
+            }
+        }
+
+        it("adds an explanation for link loops") {
+            val a = tempFolder.tmpDir.resolve("linkA")
+            val b = tempFolder.newSymbolicLink("linkB", a)
```

Comment:
I was thinking that adding a test case with a relative symlink might be a good idea as well. e.g. 
```
val c = tempFolder.tmpDir.resolve("nonExisting").resolve("..").resolve("linkA")
val b = tempFolder.newSymbolicLink("linkB", c)
```

Reply:
Done.